### PR TITLE
[15.0][REF] change name l10n_th_sequence_refactored -> l10n_th_base_sequence

### PR DESCRIPTION
**Pull request by Saran440**
1. Change name `l10n_th_sequence_refactored` to `l10n_th_base_sequence`
2. Combine 4 module into 1 module
    - l10n_th_sequence_refactored
    - l10n_th_sequence_range_date
    - l10n_th_sequence_qoy
    - l10n_th_sequence_be

### CODE REVIEW

**Review by Nantikan23 — APPROVED** [review]
Functional Tested

File: l10n_th_base_sequence/__manifest__.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
 
 {
-    "name": "Base Sequence",
+    "name": "Thailand - Base Sequence",
```
Comment (ps-tubtim) [inline l10n_th_base_sequence/__manifest__.py]:
```suggestion
    "name": "Thai Localization - Base Sequence",
```
I think you should use **Thai Localization** instead. What do you think?

File: l10n_th_base_sequence/models/ir_sequence.py [hunk 2]
```diff
@@ -62,12 +62,48 @@ def _interpolation_dict(self, date=None, date_range=None):
             "min": "%M",
             "sec": "%S",
         }
+
         res = {}
+        range_end_date = fields.Datetime.from_string(
+            self._context.get("ir_sequence_date_range_end")
+        )
+        xx = effective_date.strftime("%Y-%m-%d")
+        datetime.strptime(xx, "%Y-%m-%d")
```
Comment (ps-tubtim) [inline l10n_th_base_sequence/models/ir_sequence.py]: Variable names should be descriptive.

**Comment by Saran440** [comment]
@ps-tubtim Thanks for review. I remove field is not used and change name to makesense. Can you review again please?

**Review by ps-tubtim — APPROVED** [review]

**Pull request merged**